src/Tooltip/Tooltip.tsx

```diff
@@ -8,113 +11,198 @@ export interface TooltipMessageGroupItem {
     message: string;
 }
 
-interface TooltipState {
-    toggle: boolean;
-}
-
+export type TooltipTrigger = "hover" | "click" | "focus";
 export type TooltipTheme = "default" | "light" | "primary" | "warning" | "success" | "danger" | "purple";
-export type TooltipPosition = "top" | "bottom" | "left" | "right" | "top-right" | "top-left" | "bottom-right" | "bottom-left" | "left-top" | "left-bottom" | "right-top" | "right-bottom";
+export type TooltipPosition = ElementPosition;
 
 export interface TooltipProps {
     className?: string;
     customSvg?: React.ReactNode;
     id?: string;
+    ref?: React.LegacyRef<Tooltip>;
```

Comment:
has this been removed yet?

Reply:
Also don't forget to remove from the README file as well if it mentions ref

Reply:
this ref is needed for forceShow function